modules/fedimint-ln/src/lib.rs

```diff
@@ -275,6 +276,7 @@ impl FederationModule for LightningModule {
         let account_db_key = ContractKey(input.contract_id);
         let mut contract_account = self
             .db
+            .begin_transaction()
```

Comment:
I know this replicates old behavior, but shouldn't we just switch to only reading from the DB tx and actually remove the DB reference from the module all together?

Reply:
The PR actually removes all other functions from the trait besides begin_transaction(). So the only thing to do with a database reference is to create a new transaction.

The rocksdb implementation needs a database reference to create the transaction. I'm open to suggestions if you have ideas on encapsulating this (perhaps creating a wrapper like this: self.create_transaction()....).